File: clearclaim-frontend/src/models/Insuranceplan.ts
// Insuranceplan.ts
// Represents the InsurancePlan entity from the backend
// Maps to the insuranceplan table in PostgreSQL

export interface Insuranceplan {
  planId: number;
  planName: string;
  premiumAmount: number;
  coverageAmount: number;
  maxMembers: number;
  policyDuration: number;
}

File: Medical-Insurance/Com.Application.Domain.Entities/Insuranceplan.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace Com.Application.Domain.Entities;

public partial class Insuranceplan : BaseEntity
{
    public int PlanId { get; set; }

    public string PlanName { get; set; } = null!;

    public decimal PremiumAmount { get; set; }

    public decimal CoverageAmount { get; set; }

    public int MaxMembers { get; set; }

    public int? PolicyDuration { get; set; }
    [JsonIgnore]
    public virtual ICollection<Policy> Policies { get; set; } = new List<Policy>();
    [JsonIgnore]
    public virtual ICollection<Hospital> Hospitals { get; set; } = new List<Hospital>();
}
